src/AggregateFunctions/Streaming/AggregateFunctionDistinct.h

```diff
@@ -19,84 +28,77 @@ struct AggregateFunctionDistinctSingleNumericData
     using Self = AggregateFunctionDistinctSingleNumericData<T>;
     Set set;
 
-    /// proton: starts. Resolve multiple finalizations problem for streaming global aggreagtion query
+    /// Resolve multiple finalizations problem for streaming global aggreagtion query
+    /// Optimized, put the new coming data that the set does not have into extra_data_since_last_finalize.
     std::vector<T> extra_data_since_last_finalize;
-    bool use_extra_data = false;  /// Optimized, only streaming global aggreagtion query need to use extra data after first finalization.
-    /// proton: ends.
 
+    std::vector<uintptr_t> related_places;
+
+    /// not used
+    bool use_extra_data = false;
     void add(const IColumn ** columns, size_t /* columns_num */, size_t row_num, Arena *)
     {
         const auto & vec = assert_cast<const ColumnVector<T> &>(*columns[0]).getData();
-        /// proton: starts.
         auto [_, inserted] = set.insert(vec[row_num]);
-        if (use_extra_data && inserted)
+        if (inserted)
+        {
             extra_data_since_last_finalize.emplace_back(vec[row_num]);
-        /// proton: ends.
+        }
     }
 
-    void merge(const Self & rhs, Arena *)
+    void merge(const Self & rhs, Arena *, ConstAggregateDataPtr __restrict place)
```

Comment:
don't need pass in `ConstAggregateDataPtr __restrict place`,
we can use `reinterpret_cast<uintptr_t>(&rhs)` instead